src/syscall/systemio.h

```diff
@@ -184,18 +189,16 @@ class SystemIO : public QObject {
       });
     }
 
-    // Determine whether a given fd is already in use with the given flag.
-    static bool fdInUse(int fd, int flag) {
+    // Determine whether a given fd is already in use with either Read or Write permissions
+    static bool fdInUse(int fd, Permission perm) {
       if (fd < 0 || fd >= SYSCALL_MAXFILES) {
         return false;
       } else if (fileNames[fd].isEmpty()) {
         return false;
-      } else if ((fileFlags[fd] & flag) ==
-                 static_cast<unsigned>(
-                     flag) /* also compares ie. O_RDONLY (0x0) */) {
-        return true;
+      } else if (perm == READ_PERMS) {
+        return (fileFlags[fd] + 1) & 1; // LSB is 1 if the fd has read perms
       }
-      return false;
+      return (fileFlags[fd] + 1) & 2; // 2nd LSB is 1 if the fd has write perms
```

Comment:
Mixing arithmetic and bitwise logic is a bit too funky for me. I think these expressions can be rewritten using bit extracts and XORs.

Reply:
secondly, we should also extract out the _nth LSB is X perms_ to an enum, e.g. something like
```c++
enum FilePermissions {
   RD_PERM_FLAG = 1 << 0,
   WR_PERM_FLAG = 1 << 1
};
```